### PR TITLE
Login / Create account – Web

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

**Comment by alisa911** [comment]
https://github.com/osmandapp/OsmAnd-Issues/issues/2635

File: map/src/menu/login/ChangeResetPwd.jsx [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+        event.preventDefault();
+    };
+
+    const handleClickShowPassword2 = () => {
+        setShowPassword2(!showPassword2);
+    };
+
+    const handleMouseDownPassword2 = (event) => {
+        event.preventDefault();
+    };
```
Comment (RZR-UA) [inline map/src/menu/login/ChangeResetPwd.jsx:82]: please use 1-liners even without { } for the real one-liners:

const handleCodeChange = (e) => setCode(e.target.value);

File: map/src/menu/login/ChangeResetPwd.jsx [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+        if (emailError && emailError !== EMPTY_INPUT) {
+            setOpenCodeInput(false);
+        }
+    }, [emailError]);
+
+    useEffect(() => {
+        if (userPassword1 !== EMPTY_INPUT && userPassword2 !== EMPTY_INPUT) {
+            if (userPassword1 !== userPassword2) {
+                setPasswordError(t('web:passwords_not_match'));
+            } else if (userPassword1.length < 8 || userPassword2.length < 8) {
```
Comment (RZR-UA) [inline map/src/menu/login/ChangeResetPwd.jsx:94]: Hmm.
Code duplicate with usePasswordValidation hook?
Password rules (such as 8 chars minimum) should be placed in one validation method.

File: map/src/menu/login/CreateAccount.jsx [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+    const [userPassword2, setUserPassword2] = useState(EMPTY_INPUT);
+    const [showPassword1, setShowPassword1] = useState(false);
+    const [showPassword2, setShowPassword2] = useState(false);
+    const [createAccountError, setCreateAccountError] = useState(EMPTY_INPUT);
+
+    const passwordError = usePasswordValidation(userPassword1, userPassword2);
+
+    const handleClickShowPassword1 = () => {
+        setShowPassword1(!showPassword1);
+    };
```
Comment (RZR-UA) [inline map/src/menu/login/CreateAccount.jsx:43]: Use 1-liners please.

**Comment by RZR-UA** [comment]
Additionally, please test:

1) DELETE_ACCOUNT_URL (load URL without login)
2) DELETE_ACCOUNT_URL (load URL after login)
3) LOGIN_LOGOUT_URL (logout by server)

File: tests/selenium/src/tests/84-logout-by-server.mjs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+export default async function test() {
+    await actionOpenMap();
+    await actionLogIn();
+
+    await waitBy(By.id('se-logout-button'));
+
+    console.log('Logout by server');
+    await driver.manage().deleteAllCookies();
+    console.log('Cookies deleted');
```
Comment (RZR-UA) [inline tests/selenium/src/tests/84-logout-by-server.mjs:16]: Please remove console.log

**Review by RZR-UA — APPROVED** [review]

**Review by RZR-UA — APPROVED** [review]

**Pull request merged**